hack/setup_env.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
```

Comment:
Even though it's a build environment, I'm not sure I like it, taking logic and putting it as one chunk of image layer in docker
It's also kind of another build environment implemented, and we have no control of the prerequisites of this environment

Why not using podman for anywhere we need this?

Reply:
Because Docker in Docker isn't supported on Prow

Reply:
As I wrote in the description:
> Adding a setup_env.sh script for users that don't use Dockerfile to run
the Makefile targets, for example our Prow CI.